src/eth_rpc/mod.rs

```diff
@@ -39,15 +49,33 @@ pub async fn run_server(
     let http_middleware =
         tower::ServiceBuilder::new().layer(ProxyGetRequestLayer::new("/health", "net_health")?).layer(cors);
 
+    let registry = Registry::new();
+    let metrics = RpcMetrics::new(Some(&registry))?.map(|m| MetricsLayer::new(m, "http"));
+    tokio::spawn(async move {
+        let _ = init_prometheus(
+            SocketAddr::new(
+                std::net::IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
+                get_env_or_default("PROMETHEUS_PORT", "9615").parse().unwrap(),
+            ),
+            registry.clone(),
+        )
+        .await;
+    });
+    let rpc_middleware = RpcServiceBuilder::new().option_layer(metrics);
```

Comment:
Can you explain for me what you are doing here? Also is it ok to clone the registry or should the init_prometheus get a reference to it?

Reply:
Added comments. Good catch, it works without the clone!